PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = ['purchase.order']
-    _inherit = ['purchase.order.line']
-    _inherit = ['purchase.requisition']
-    _inherit = ["purchase.requisition.line"]
-    _inherit = ['stock.rule']
-    _inherit = ['stock.move']

Total Changes: 12

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/purchase_requisition_stock/models/purchase.py
+++ b/addons/purchase_requisition_stock/models/purchase.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class PurchaseOrder(models.Model):
-    _inherit = ['purchase.order']
+    _inherit = 'purchase.order'
 
     on_time_rate_perc = fields.Float(string="OTD", compute="_compute_on_time_rate_perc")
 
@@ -25,6 +25,6 @@ class PurchaseOrder(models.Model):
 
 
 class PurchaseOrderLine(models.Model):
-    _inherit = ['purchase.order.line']
+    _inherit = 'purchase.order.line'
 
     on_time_rate_perc = fields.Float(string="OTD", related="order_id.on_time_rate_perc")

--- a/addons/purchase_requisition_stock/models/purchase_requisition.py
+++ b/addons/purchase_requisition_stock/models/purchase_requisition.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class PurchaseRequisition(models.Model):
-    _inherit = ['purchase.requisition']
+    _inherit = 'purchase.requisition'
 
     def _default_picking_type_id(self):
         picking_type = self.env['stock.picking.type'].search([('warehouse_id.company_id', '=', self.env.company.id), ('code', '=', 'incoming')], limit=1)
@@ -20,7 +20,7 @@ class PurchaseRequisition(models.Model):
 
 
 class PurchaseRequisitionLine(models.Model):
-    _inherit = ["purchase.requisition.line"]
+    _inherit = "purchase.requisition.line"
 
     move_dest_id = fields.Many2one('stock.move', 'Downstream Move')
 

--- a/addons/purchase_requisition_stock/models/stock.py
+++ b/addons/purchase_requisition_stock/models/stock.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class StockRule(models.Model):
-    _inherit = ['stock.rule']
+    _inherit = 'stock.rule'
 
     def _prepare_purchase_order(self, company_id, origins, values):
         res = super(StockRule, self)._prepare_purchase_order(company_id, origins, values)
@@ -26,7 +26,7 @@ class StockRule(models.Model):
 
 
 class StockMove(models.Model):
-    _inherit = ['stock.move']
+    _inherit = 'stock.move'
 
     requisition_line_ids = fields.One2many('purchase.requisition.line', 'move_dest_id')
 
